userprofile/tests/test_views.py

```diff
@@ -430,6 +427,20 @@ def test_existing_profile(self):
         self.assertEqual(response.data["msisdn"], "+27820001001")
         self.assertEqual(response.data["first_name"], "testname")
 
+    def test_update_profile(self):
+        HealthCheckUserProfile.objects.create(
+            msisdn="+27820001001", first_name="testname"
+        )
+        user = get_user_model().objects.create_user("test")
+        user.user_permissions.add(
+            Permission.objects.get(codename="change_healthcheckuserprofile")
+        )
+        self.client.force_authenticate(user)
+        response = self.client.patch(self.url, {"data": {"something": "updated"}})
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["data"]["something"], "updated")
```

Comment:
We should probably check that the actual object in the database was updated.

Also, we have to be careful about replacing the entire `data` blob here when we call this endpoint. I don't know if it's easy to rather make this a merge rather than replace.